share/functions/fish_svn_prompt.fish

```diff
@@ -70,15 +70,15 @@ set -g __fish_svn_prompt_char_token_broken_color --bold magenta
 function __fish_svn_prompt_parse_status --argument flag_status_string --description "helper function that does pretty formatting on svn status"
     # SVN status symbols
     # Do not change these! These are the expected characters that are output from `svn status`, they are taken from `svn help status`
-    set -l __fish_svn_prompt_chars A C D I M R X \? ! ~ L + S K O T B
+    set -l __fish_svn_prompt_chars A C D I M R X \\\? ! \~ L + S K O T B
     # this sets up an array of all the types of status codes that could be returned.
-    set -l __fish_svn_prompt_flag_names added conflicted deleted ignored modified replaced unversioned_external unversioned missing locked scheduled switched token_present token_other token_stolen token_broken
+    set -l __fish_svn_prompt_flag_names added conflicted deleted ignored modified replaced unversioned_external unversioned missing versioned_obstructed locked scheduled switched token_present token_other token_stolen token_broken
     # iterate over the different status types
     for flag_type in $__fish_svn_prompt_flag_names
         # resolve the name of the variable for the character representing the current status type
         set -l flag_index (contains -i $flag_type $__fish_svn_prompt_flag_names)
         # check to see if the status string for this column contains the character representing the current status type
-        if test (count (string match $__fish_svn_prompt_chars[$flag_index] -- $flag_status_string)) -eq 1
+        if test (count (string match -e $__fish_svn_prompt_chars[$flag_index] -- $flag_status_string)) -eq 1
```

Comment:
I don't see how this works?

$flag_status_string is received from

```fish
__fish_svn_prompt_parse_status $column_status
```

And $column_status is one element (because of `tr -d ' \n'`), so this would only ever succeed if this is the only status character.

That doesn't seem correct?

Reply:
Um, no. $column_status can be, for example, 'AM', if you have one file added and one file modified.
This will effectively result in `string match -e A -- AM` and `string match -e M -- AM`, which will both succeed. Without `-e`, they will not. In both cases the number of matches is 1 (although checking the count seems a bit unnecessary, but I wanted to keep my changes minimal).

Reply:
Ah, yeah, right, "--entire" *adds* the wildcards at both ends.

I always forget the semantics of string's glob mode (and tbh we should have just gone full-regex).